Detectors/TRD/simulation/src/Digitizer.cxx

```diff
@@ -96,19 +96,21 @@ void Digitizer::flush(DigitContainer& digits, o2::dataformats::MCTruthContainer<
         LOG(WARN) << "TRD conversion of signals to digits failed";
       }
       for (const auto& iter : smc) {
-        labels.addElements(labels.getIndexedSize(), iter.second.labels);
+        if (iter.second.isDigit)
+          labels.addElements(labels.getIndexedSize(), iter.second.labels);
       }
     }
   } else {
     // since we don't have any pileup signals just flush the signals for each chamber
     // we avoid flattening the array<map, ndets> to a single map
-    for (const auto& smc : mSignalsMapCollection) {
+    for (auto& smc : mSignalsMapCollection) {
```

Comment:
why do you need to take away the const?

Reply:
The `smc` container gets modified when converting signals to adcs, just a couple lines below the for-loop start. There, the signals that really become digits are flagged (hence the container needs to be non-const) so that only signals ending up in digits will pass the MC labels t the output label container.